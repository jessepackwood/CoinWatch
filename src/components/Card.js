import React from 'react'

const Card = () => {
  return(
    <div>
      <h3></h3>
      <span></span>
    </div>
  )
}