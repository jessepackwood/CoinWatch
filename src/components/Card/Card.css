@import url("https://fonts.googleapis.com/css?family=Abel");
.card {
  border: 1px solid red;
  margin: 10px auto; }

.coin-name {
  color: orange;
  font-family: "Abel", sans-serif; }
