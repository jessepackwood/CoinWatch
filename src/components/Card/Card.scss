@import '../../variable.scss';

.card {
	margin: 15px auto;
	display: flex;
	padding: 10px;
	// border-radius: 6px;
	border-bottom: 1px solid white;
	background: $background;
}
.number {
	margin-right: 5px;
	color: #fff;
}
.coin-name {
	font-size: 22pt;
	color: $orange;
	font-family: $title-font;
	margin: 0 10px;
	width: 350px;
	&.port-coin-name {
		width: 200px;
		color: $orange;
	}
}

.short {
	margin-left: 5px;
	color: #fff;
}

.price {
	color: white;
	font-size: 20pt;
	margin: 0 auto;
	width: 200px;
	&.port-price {
	margin: 0;
	}
}

.dayChange {
	text-align: right;
	margin-right: 15px;
	font-size: 20pt;
	width: 200px;
}

.positive {
	color: #9af765;
}

.negative {
	color: rgb(245, 84, 70);
}

.hour {
	margin-left: 10px;
	color: white;
}

.amount {
	font-size: 20pt;
	margin: 0 auto;
	color: #fff;
	width: 200px;
}

.btn-fav {
	width: 35px;
	height: 35px;
	background-image: url('../../assets/rounded-add-button.svg');
	background-size: contain;
	background-repeat: no-repeat;
	&.minus {
		&:hover {
			background-image: url('../../assets/tick-inside-circle-red.svg');
		}
	background-image: url('../../assets/tick-inside-circle.svg');
	background-repeat: no-repeat;
	}
	&:hover {
		cursor: pointer;
		background-image: url('../../assets/rounded-add-green.svg')
	}
}

.port-btn {
	margin-left: 10px;
}

.usd {
	margin-right: 3px;
}
