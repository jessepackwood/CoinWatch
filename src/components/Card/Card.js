import React from 'react'
import './Card.css'
import { connect } from 'react-redux'
// import { addToWatchList } from '../../services/coinCapServices.js'
import * as actions from '../../actions'

const Card = ({ name, short, dayChange, price, number, addCoinToWatch, removeCoinFromWatch, coin }) => {

	const addColor =  dayChange > 0 ? 'positive' : 'negative'

  return(
    <div className={`${addColor} card`}>
    	
      <h3 className='coin-name'>
      	<span className='number'>{number}.</span>
      {name}
      	<span className='short'>({short})</span>
      </h3>
      <span className='price'>${price}</span>
      <span className={`${addColor} dayChange`}>{dayChange}%</span>
      <button className='btn-fav' onClick={() => addCoinToWatch(coin)}>Watch</button>
    </div>
  )
}

export const mapStateToProps = state => ({
	coin: state.coins[0]
})

export const mapDispatchToProps = dispatch => {
	return {
		addCoinToWatch: (coin) => {
			dispatch(actions.addWatch(coin))
		},
		removeCoinFromWatch: (coin) => {
			dispatch(actions.removeWatch(coin))
		}
	}
}
export default connect(mapStateToProps, mapDispatchToProps)(Card);