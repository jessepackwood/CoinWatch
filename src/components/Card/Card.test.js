import React from 'react'
import { shallow } from 'enzyme';
import { Card, mapStateToProps, mapDispatchToProps } from './Card';

describe('Card tests', () => {

  const mockAddWatch = jest.fn()
  const mockRemoveWatch = jest.fn()

  let card;
  let coin;

  beforeEach( () => {
    coin = {long: 'Bitcoin'}
    card = shallow(<Card coin={coin}/>)
  });

  it('should be defined', () => {
    expect(card).toBeDefined()
  })

  it('should match the snapshot', () => {
    expect(card).toMatchSnapshot();
  });

  it('should call addWatch on click', () => {
    card.find('.btn-fav').first().simulate('click');
    expect(mockAddWatch).toHaveBeenCalled();
  });

  it('should call removeWatch on click', () => {

    card.find('.btn-fav').first().simulate('click');
    expect(mockRemoveWatch).toHaveBeenCalled();
  });
})