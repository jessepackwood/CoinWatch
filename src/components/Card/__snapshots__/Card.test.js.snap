// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests should match the snapshot 1`] = `undefined`;
